### ISSUE TITLE
bcg729 outdated version

### ISSUE THREAD

**Issue by joelsdc** [xadhoom/mod_bcg729#14]
Hi, 

The Makefile is pointing to v1.0.1 of bcg729 (from Nov-2015), in the src repo there is a v1.0.2 (from Aug-2016).

```
root:/opt/bcg729# git tag
0.2
1.0.0
1.0.1
1.0.2
root:/opt/bcg729#
```

Any specific reason on why you haven't updated to use v1.0.2?

Thanks!

Best regards, 
Joel

**Comment by p00h** [comment]
There is no any specific reason, this is opensource code ;) Speaking about
agenda, I've just updated to latest fs from upstream. I haven't found any
problems since i don't use G711. I can confirm there is two way audio with
alaw, ulaw, speex, opus transcoding. Have you seen any changes according to
G711 code at fs change log?

**Comment by xadhoom** [comment]
Simply because the bcg729 api have changed in new versions and I don't have time to look at them and do the proper updates in the FS codec.
But I'll be glad to review any PR :)

**Comment by ti24horas** [comment]
@joelsdc were you able to compile g729 for freeswitch (version 1.0.3 with CNG and VAD)?

**Comment by MartB** [comment]
Not sure if it includes all the important stuff, but calling internally worked fine between two linphone clients.
https://github.com/xadhoom/mod_bcg729/pull/18

**Comment by xadhoom** [comment]
please test as @MartB PR has just been merged.

**Comment by MartB** [comment]
Thanks for that!

**Comment by madovsky** [comment]
but now it complains on Fedora 24 64bit


gcc  -fPIC -O3 -fomit-frame-pointer -fno-exceptions -Wall -std=c99 
-pedantic -m64 -mtune=generic -I/usr/include -Ibcg729/include 
-I/usr/local/freeswitch/include/freeswitch -c mod_bcg729.c
mod_bcg729.c: In function ‘switch_bcg729_encode’:
mod_bcg729.c:107:13: error: too many arguments to function ‘bcg729Encoder’
              bcg729Encoder(context->encoder_object, ddp, edp, &frameSize);
              ^~~~~~~~~~~~~
In file included from mod_bcg729.c:31:0:
bcg729/include/bcg729/encoder.h:57:24: note: declared here
  BCG729_VISIBILITY void bcg729Encoder(bcg729EncoderChannelContextStruct 
*encoderChannelContext, int16_t inputFrame[], uint8_t bitStream[]);
                         ^~~~~~~~~~~~~
mod_bcg729.c: In function ‘switch_bcg729_decode’:
mod_bcg729.c:145:57: warning: passing argument 4 of ‘bcg729Decoder’ 
makes pointer from integer without a cast [-Wint-conversion]
          bcg729Decoder(context->decoder_object, NULL, 0, 1, 0, 0, ddp);
                                                          ^
In file included from mod_bcg729.c:32:0:
bcg729/include/bcg729/decoder.h:57:24: note: expected ‘int16_t * {aka 
short int *}’ but argument is of type ‘int’
  BCG729_VISIBILITY void bcg729Decoder(bcg729DecoderChannelContextStruct 
*decoderChannelContext, uint8_t bitStream[], uint8_t frameErasureFlag, 
int16_t signal[]);
                         ^~~~~~~~~~~~~
mod_bcg729.c:145:9: error: too many arguments to function ‘bcg729Decoder’
          bcg729Decoder(context->decoder_object, NULL, 0, 1, 0, 0, ddp);
          ^~~~~~~~~~~~~
In file included from mod_bcg729.c:32:0:
bcg729/include/bcg729/decoder.h:57:24: note: declared here
  BCG729_VISIBILITY void bcg729Decoder(bcg729DecoderChannelContextStruct 
*decoderChannelContext, uint8_t bitStream[], uint8_t frameErasureFlag, 
int16_t signal[]);
                         ^~~~~~~~~~~~~
mod_bcg729.c:156:9: error: too many arguments to function ‘bcg729Decoder’
          bcg729Decoder(context->decoder_object, edp, encoded_data_len, 
0, isSID, 0, ddp);
          ^~~~~~~~~~~~~
In file included from mod_bcg729.c:32:0:
bcg729/include/bcg729/decoder.h:57:24: note: declared here
  BCG729_VISIBILITY void bcg729Decoder(bcg729DecoderChannelContextStruct 
*decoderChannelContext, uint8_t bitStream[], uint8_t frameErasureFlag, 
int16_t signal[]);
                         ^~~~~~~~~~~~~
Makefile:26: recipe for target 'mod_bcg729.o' failed
make: *** [mod_bcg729.o] Error 1

**Comment by xadhoom** [comment]
mmmh have you cleaned up the bcg729 dir in order to force a fresh checkout ?

**Comment by madovsky** [comment]
yes it's what I did just now, my git clone was somehow corrupted.
I made fresh cloning and so good so far...

**Comment by MartB** [comment]
I tested it a little bit further works great with my voicemail and generic ivr stuff.

**Comment by joelsdc** [comment]
My tests all work nicely... Thanks @MartB ! Should I close the issue? I don't know if the comment from @ROBERT-MCDOWELL applies to this change or should be checked independently..

**Comment by MartB** [comment]
should be okay to close

**Comment by MartB** [comment]
should be okay to close

**Comment by xadhoom** [comment]
Merged #24 , thanks!
